proactive-healthcheck/api-health-check.sh

```diff
@@ -0,0 +1,604 @@
+#!/bin/bash
+#set -x
+
+#input parameters
+#username to connect to the ProActive Portals
+username=<CHANGE_ME>
+#password of the user connecting to the ProActive Portals
+proactiveAdminPasswordDecrypted=<CHANGE_ME>
+#server DNS name or public IP
+serverDNSName=try.activeeon.com
+#server Port
+webPort=8443
+#web protocol, http or https
+webProtocol=https
+#name of one of the node sources, needed in a test to get its description
+nsName=local
+#timeout in seconds to check free nodes, job submission, and job status
+timeoutInSeconds=120
+
+#set colors for printf state of services
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+GREEN='\033[0;32m'
+PURPLE='\033[0;35m'
+
+#number of failed tests
+FAIL=0
+#number of successful tests
+SUCCESS=0
+
+#complete ProActive server web URL
+URL="${webProtocol}://${serverDNSName}:${webPort}"
+
+#tracks the number of seconds that have passed since the tests were started
+SECONDS=0
+
+#get the session id, needed for all tests
+endpoint=$(curl -o - -d "username=${username}&password=${proactiveAdminPasswordDecrypted}" --write-out "\n%{http_code}\n" --silent --output /dev/null ${URL}/rest/scheduler/login)
+response=($endpoint)
+sessionid=${response[0]}
+status=${response[1]}
+message="$NC: ${PURPLE}(POST: scheduler/login)$NC GET A SESSION ID"
+if [ "$status" == "200" ]; then
+  ((SUCCESS++))
+  echo -e "${GREEN} OK $message"
+else
+  echo -e "${RED} NOK $message"
+  ((FAIL++))
+  #exit the script if the session id could not be retrieved since it is required by most of the tests
+  exit 1
+fi
+
+#function used by tests to check the returned status and print messages
+check_endpoint() {
+  endpoint=$1
+  message=$2
+  if [[ "$endpoint" == "200" || "$endpoint" == "201" ]]; then
+    echo -e "${GREEN} OK ${message}"
+    ((SUCCESS++))
+  else
+    echo -e "${RED} NOK ${message}"
+    ((FAIL++))
+  fi
+}
+
+#wait until a node becomes free before starting the tests
+counter=0
+nodeState=""
+until [[ $nodeState == "FREE" ]]; do
+  monitoring=$(curl -G --header "sessionid:$sessionid" --silent "$URL/rest/rm/monitoring/")
+  nodeEvent=$(echo $monitoring | jq -r ".nodesEvents[0]")
+  nodeState=$(echo $nodeEvent | jq -r ".nodeState")
+  sleep 1
+  ((counter++))
+  if [ "$counter" == "$timeoutInSeconds" ]; then
```

Comment:
I agree with Ali, let replace **timeoutInSeconds** with **checks** or **attempts**

Reply:
it is an input parameter to the script it should be a meaningful name:
maybe attempsOnFailure?

Reply:
attemp**t**sOnFailure, I prefer avoiding the word failure, I suggest **endpoint_check_attempts**